test/tools/ossfuzz/protomutators/YulProtoMutator.cpp

```diff
@@ -0,0 +1,2081 @@
+	}
+);
+
+// Remove multi variable declaration
+static YPR<Block> removeMultiVarDecl(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				for (auto &stmt: *_message->mutable_statements())
+					if (stmt.has_multidecl())
+					{
+						delete stmt.release_multidecl();
+						stmt.clear_multidecl();
+						break;
+					}
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Remove multi variable declaration from statement block"
+		);
+	}
+);
+
+// Add function definition
+static YPR<Block> addFuncDef(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator& _rand)
+			{
+				auto funcDef = new FunctionDef();
+				funcDef->set_num_input_params(_rand());
+				funcDef->set_num_output_params(_rand());
+				funcDef->set_allocated_block(new Block());
+				// TODO: Add assignments to output params if any
+				_message->add_statements()->set_allocated_funcdef(funcDef);
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Add function definition to statement block"
+		);
+	}
+);
+
+// Remove function definition
+static YPR<Block> removeFuncDef(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				for (auto &stmt: *_message->mutable_statements())
+					if (stmt.has_funcdef())
+					{
+						delete stmt.release_funcdef();
+						stmt.clear_funcdef();
+						break;
+					}
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Remove function definition from statement block"
+		);
+	}
+);
+
+// Add bounded for stmt
+static YPR<Block> addBoundedFor(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				_message->add_statements()->set_allocated_boundedforstmt(new BoundedForStmt());
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Add bounded for statement to statement block"
+		);
+	}
+);
+
+// Remove bounded for stmt
+static YPR<Block> removeBoundedFor(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				for (auto &stmt: *_message->mutable_statements())
+					if (stmt.has_boundedforstmt())
+					{
+						delete stmt.release_boundedforstmt();
+						stmt.clear_boundedforstmt();
+						break;
+					}
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Remove bounded for statement from statement block"
+		);
+	}
+);
+
+// Add generic for stmt
+static YPR<Block> addGenericFor(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				_message->add_statements()->set_allocated_forstmt(new ForStmt());
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Add for statement to statement block"
+		);
+	}
+);
+
+// Remove generic for stmt
+static YPR<Block> removeGenericFor(
```

Comment:
Could it happen that calling this function removes a bounded for? Is that ok or not intended?

Reply:
No. Bounded and generic for are different protobuf messages. This only removes a generic for.

However, it could happen that a generic for actually encodes a bounded for (since it is a general for statement in fact). If this happens, it is okay.